src/Stepper/index.js

```diff
@@ -4,79 +4,71 @@ import * as React from "react";
 import StepperStateless from "./StepperStateless";
 import validateIncrement from "../utils/validateIncrement";
 import validateDecrement from "../utils/validateDecrement";
+import useStateWithCallback from "../hooks/useStateWithCallback";
 
-import type { Props, State } from "./index";
+import type { Props } from "./index";
 
-class Stepper extends React.PureComponent<Props, State> {
-  state = {
-    value: this.props.defaultValue || 0,
-  };
-
-  setValueAndInjectCallback = (value: number) => {
-    const { onChange } = this.props;
+const Stepper = ({ onChange, defaultValue, ...props }: Props) => {
+  const [value, setValue] = useStateWithCallback<number>(defaultValue || 0, val => {
     if (onChange) {
-      onChange(value);
+      onChange(val);
     }
-    this.setState({ value });
-  };
 
-  incrementCounter = () => {
-    const { value } = this.state;
-    const { maxValue = Number.POSITIVE_INFINITY, step = 1 } = this.props;
+    setValue(val);
+  });
 
-    this.setValueAndInjectCallback(validateIncrement({ value, maxValue, step }));
-  };
+  const incrementCounter = React.useCallback(() => {
+    const { maxValue = Number.POSITIVE_INFINITY, step = 1 } = props;
 
-  decrementCounter = () => {
-    const { value } = this.state;
-    const { minValue = Number.NEGATIVE_INFINITY, step = 1 } = this.props;
+    setValue(validateIncrement({ value, maxValue, step }));
+  }, [props, setValue, value]);
```

Comment:
I understand the reason why you are desctructurizing the props in the callback directly, but using `props` inside dependency array would cause to re-creation of the callback upon every change of the `props` object (even when e.g. `dataTest` will change). Can you look into it and find another way how to do it?